app/src/components/trace/TraceTree.tsx

```diff
@@ -62,49 +91,80 @@ function SpanTreeItem<TSpan extends ISpanItem>(props: {
           });
         }}
       >
-        <SpanNodeWrap isSelected={selectedSpanNodeId === node.span.id}>
+        <SpanNodeWrap
+          isSelected={selectedSpanNodeId === node.span.id}
+          nestingLevel={nestingLevel}
+        >
           <Flex
             direction="row"
             gap="size-100"
             justifyContent="start"
             alignItems="center"
+            flex="1 1 auto"
+            minWidth={0}
           >
             <SpanKindIcon spanKind={node.span.spanKind} />
-            <SpanItem {...node.span} />
+            <span css={spanNameCSS} title={name}>
+              {name}
+            </span>
+            {statusCode === "ERROR" ? (
+              <SpanStatusCodeIcon statusCode="ERROR" />
+            ) : null}
+            {typeof tokenCountTotal === "number" ? (
+              <TokenCount
+                tokenCountTotal={tokenCountTotal}
+                tokenCountPrompt={tokenCountPrompt ?? 0}
+                tokenCountCompletion={tokenCountCompletion ?? 0}
+              />
+            ) : null}
+            {latencyMs === null ? null : (
+              <LatencyText latencyMs={latencyMs} showIcon={false} />
+            )}
           </Flex>
+          <div css={spanControlsCSS} data-testid="span-controls">
+            {hasChildren ? (
+              <CollapseToggleButton
+                isCollapsed={isCollapsed}
+                onClick={() => {
+                  setIsCollapsed(!isCollapsed);
+                }}
+              />
+            ) : null}
+          </div>
         </SpanNodeWrap>
       </button>
       {childNodes.length ? (
         <ul
-          css={(theme) => css`
-            margin: var(--ac-global-dimension-static-size-100) 0 0
-              var(--ac-global-dimension-static-size-600);
-            display: flex;
+          css={css`
+            display: ${isCollapsed ? "none" : "flex"};
             flex-direction: column;
-            gap: ${theme.spacing.padding8}px;
           `}
         >
           {childNodes.map((leafNode, index) => {
             // The last child does not need an edge connector, a line to connect the nodes
             // after to the parent node
             const nexSibling = childNodes[index + 1];
             return (
-              <div
+              <li
                 key={leafNode.span.context.spanId}
                 css={css`
                   position: relative;
                 `}
               >
                 {nexSibling ? (
-                  <SpanTreeEdgeConnector {...nexSibling.span} />
+                  <SpanTreeEdgeConnector
+                    {...nexSibling.span}
+                    nestingLevel={nestingLevel}
+                  />
                 ) : null}
-                <SpanTreeEdge {...leafNode.span} />
+                <SpanTreeEdge {...leafNode.span} nestingLevel={nestingLevel} />
```

Comment:
why is the edges level not incremented? because the connection between the two spans should be a the current spans level not the next?

Reply:
Because it's the edge of the curve at the current level. It's the little rounding on the left of the icon